### ISSUE TITLE
add a removeAllKeys or similar

### ISSUE THREAD

**Issue by renatosc** [jrendel/SwiftKeychainWrapper#28]
First, congrats for the wrapper. Simple and efficient.

I just missed out one function:  a removeAllKeys()  function that would erase all Keychain items. That is very useful to clean the Keychain on the app first run.

**Comment by crashoverride777** [comment]
you need to add these 2 methods to do so.

/// Remove all keychain data
    class func removeAllData() {
        deleteKeychainSecClass(kSecClassGenericPassword)
        deleteKeychainSecClass(kSecClassInternetPassword)
        deleteKeychainSecClass(kSecClassCertificate)
        deleteKeychainSecClass(kSecClassKey)
        deleteKeychainSecClass(kSecClassIdentity)
    }

/// Delete keychain sec class
    private class func deleteKeychainSecClass(secClass: AnyObject) -> Bool {
        let query = [ kSecClass as String : secClass ]
        let status: OSStatus = SecItemDelete(query as CFDictionaryRef)
        
        if status == errSecSuccess {
            return true
        } else {
            return false
        }
    }

**Comment by JuanPabloBoero** [comment]
Improved it : 

```
class func clearAllKeychain() -> Bool {
        if deleteKeychainSecClass(kSecClassGenericPassword) && deleteKeychainSecClass(kSecClassInternetPassword) && deleteKeychainSecClass(kSecClassCertificate) && deleteKeychainSecClass(kSecClassKey) && deleteKeychainSecClass(kSecClassIdentity){
            
            return true
            
        }else{
            
            return false
            
        }
    }
    
    
    private class func deleteKeychainSecClass(secClass: AnyObject) -> Bool {
        let query = [ kSecClass as String : secClass ]
        let status = SecItemDelete(query as CFDictionaryRef)
        
        return status == errSecSuccess
        
    }
```

**Comment by crashoverride777** [comment]
Hey,

Thanks for the post. I was actually considering returning a bool for the deleteMethod as well, however in my case I would only use this method for testing so I didnt bother.
Hopefully the author puts this in his awesome helper.

Thanks again.

**Comment by jrendel** [comment]
Hey guys, I know you've been asking for this. I'm trying to find some time to work on the library, I've got a handful of changes I'd like to put into an update.

**Comment by crashoverride777** [comment]
Awesome, looking forward to it. Hope you keep it as simple and great as it is right now. The one thing I would also love to see is an option to directly save Bool and/or Int values, like you can in NSUserDefaults.  (boolForKey(...), intForKey(...) )

Thank you for your great helper.

**Comment by jrendel** [comment]
I pulled the suggested delete operation into a new branch on the repo and started testing. I have mixed feelings about a "delete all" function that has the potential to wipe more than you expected. While I agree its useful, it can also be dangerous if you don't understand what you're doing, so at the very least I want to make sure its well documented before I release it. I've also added a delete that only deletes keys that match your current wrapper settings (service name and access group).

I'm going to be continuing on with several changes for what I'm considering a "v2" of the library, and pull the delete stuff into that. As part of these changes, I will also look at adding convenience functions such as boolForKey, intForKey, etc. I always wanted this to be simple to use, like NSUserDefaults, for someone that has no understanding of keychain. Hence, why it does things like only allow access when the device is unlocked. 

At the same time, I also see the potential for users to want more control. So my approach for v2 will more closely follow the pattern established by NSUserDefaults. There will be a "StandardKeychainAccess" or something along those lines, a singleton that gives you quick and easy access with fixed settings. And then the option to create your own keychain wrapper instance with more custom settings such as ServiceName, access group and access level. For backwards compatibility, I'll leave the existing class methods on the wrapper, they will just default to using the standard instance instead of some static state. 

I've started this work in a branch called "unstable" as I tend to commit often things that I am working on but also change them often as I think of different or better ways to do something. So you're welcome to track along, but don't get too attached to anything specific :)
